sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -52,6 +53,15 @@ trait FunctionRegistry {
   /** Drop a function and return whether the function existed. */
   def dropFunction(name: String): Boolean
 
+  /** Get the builder of the specified function name and class name. */
+  def getFunctionBuilderAndInfo(
```

Comment:
By passing function name and class name, this method will return `FunctionBuilder` and `ExpressionInfo` which are used to create temporary function through `Catalog` API.